net/pidlisten/pidlisten_linux.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package pidlisten
+
+import (
+	"errors"
+	"fmt"
+	"go4.org/mem"
+	"io/fs"
+	"net"
+	"os"
+	"path/filepath"
+	"tailscale.com/util/dirwalk"
+
+	"github.com/vishvananda/netlink"
+)
+
+// NewPIDListener wraps a net.Listener so that it only accepts connections from the current process.
+func NewPIDListener(ln net.Listener) net.Listener {
+	return &listener{ln: ln}
+}
```

Comment:
I prefer the style of moving this to the main portable file (pidlisten.go) so the docs are in one place, and then having a `var newPIDListener func(net.Listener) net.Listener` that OS-specific files can init in their `init`.

Also: can this just be called `New`, given the package name?

Reply:
Oh, this part is portable. You meant to move this to the other file.